#
# We used to maintain our local changes to ltmain.sh.
#
# But it doesn't work well when a user tries to run libtoolize with
# newer libtool on his system.
#
# So, instead of keeping a change in ltmain.sh, we introduce patching
# libtool, at the last stage of configure just libtool is generated.
#
# When new libtool is released, please update this patch.  Currently,
# we assume:
#
# package_revision=2.4.7
#

--- libtool.orig	2023-07-27 15:45:41.026220580 +0900
+++ libtool	2023-07-27 11:20:52.933295428 +0900
@@ -9016,6 +9016,11 @@
 	    elif test -n "$soname_spec"; then
 	      # bleh windows
 	      case $host in
+	      x86_64-*mingw32*)
+		func_arith $current - $age
+		major=$func_arith_result
+		versuffix="6-$major"
+		;;
 	      *cygwin* | mingw* | *cegcc* | *os2*)
 	        func_arith $current - $age
 		major=$func_arith_result
@@ -9794,7 +9799,14 @@
 	  # extension on DOS 8.3 file systems.
 	  func_arith $current - $age
 	  major=$func_arith_result
-	  versuffix=-$major
+	   case $host in
+	   x86_64-*mingw32*)
+	     versuffix="6-$major"
+	     ;;
+	   *)
+	     versuffix="-$major"
+	     ;;
+	   esac
 	  ;;
 
 	*)
